src/Send/SendScreen/SendScreen.tsx

```diff
@@ -64,20 +61,17 @@ export const SendScreen = ({
   const strings = useStrings()
   const navigation = useNavigation()
   const wallet = useSelectedWallet()
-
-  const tokenBalance = useSelector(tokenBalanceSelector)
   const isFetchingBalance = useSelector(isFetchingUtxosSelector)
   const lastFetchingError = useSelector(lastUtxosFetchErrorSelector)
   const defaultAsset = getDefaultAssetByNetworkId(wallet.networkId)
+  const balance = useBalances(wallet, defaultAsset.identifier)
   const utxos = useSelector(utxosSelector)
   const hasPendingOutgoingTransaction = useSelector(hasPendingOutgoingTransactionSelector)
   const netInfo = useNetInfo()
   const isOnline = netInfo.type !== 'none' && netInfo.type !== 'unknown'
-  const selectedAsset = tokenBalance.values.find(({identifier}) => identifier === selectedTokenIdentifier)
 
-  if (!selectedAsset) {
-    throw new Error('Invalid token')
```

Comment:
The selected token identifier is updated here -> https://github.com/Emurgo/yoroi-mobile/pull/2095/files#diff-c77f2088ccc8f57d274dec697bcaa11b7eee5a8b6e320748b6f4114965d24b28R49

The prop change is propagated.